targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52832/device/TOOLCHAIN_ARM_STD/nRF52832.sct

```diff
@@ -26,22 +26,24 @@
 
 #define MBED_RAM0_START MBED_RAM_START
 #define MBED_RAM0_SIZE  0xE0
-#define MBED_RAM1_START (MBED_RAM_START + MBED_RAM0_SIZE)
+#define MBED_RAM1_START (MBED_RAM0_START + MBED_RAM0_SIZE)
 #define MBED_RAM1_SIZE  (MBED_RAM_SIZE - MBED_RAM0_SIZE)
+#define RAM_FIXED_SIZE  (MBED_CONF_TARGET_BOOT_STACK_SIZE + MBED_RAM0_SIZE)
 
 LR_IROM1 MBED_APP_START MBED_APP_SIZE {
   ER_IROM1 MBED_APP_START MBED_APP_SIZE {
    *.o (RESET, +First)
-   *(InRoot$$Sections) 
+   *(InRoot$$Sections)
    .ANY (+RO)
   }
-
   RW_IRAM0 MBED_RAM0_START UNINIT MBED_RAM0_SIZE { ;no init section
         *(*nvictable)
   }
   RW_IRAM1 MBED_RAM1_START MBED_RAM1_SIZE {
    .ANY (+RW +ZI)
   }
-  ARM_LIB_STACK MBED_RAM1_START+MBED_RAM1_SIZE EMPTY -Stack_Size { ; Stack region growing down
+  ARM_LIB_HEAP AlignExpr(+0, 16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE + MBED_RAM_START - AlignExpr(ImageLimit(RW_IRAM1), 16)) {
```

Comment:
I am getting confused by the extra variable `RAM_FIXED_SIZE`. This should be
```suggestion
  ARM_LIB_HEAP AlignExpr(+0, 16) EMPTY (MBED_RAM1_START + MBED_RAM1_SIZE - MBED_CONF_TARGET_BOOT_STACK_SIZE - AlignExpr(ImageLimit(RW_IRAM1), 16)) {
```

Reply:
* `MBED_RAM1_START + MBED_RAM1_SIZE` (same as `MBED_RAM_SIZE + MBED_RAM_START`) is the high boundary of the RAM
* `AlignExpr(ImageLimit(RW_IRAM1), 16)` is the high boundary of the application's static memory
* The area in between is shared between the stack and the heap.

The problem is that `RAM_FIXED_SIZE` contains RAM0 (vector table) which shouldn't be there - the vector table does not share space with the stack and the heap. This error exists in the scatter files of many other targets, and we'll fix it in a separate PR.